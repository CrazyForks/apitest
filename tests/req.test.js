const { spwanTest } = require("./utils");

describe("req", () => {
  test("eval", async () => {
    const { stdout, code } = await spwanTest("req/eval.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("file", async () => {
    const { stdout, code } = await spwanTest("req/file.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("main", async () => {
    const { stdout, code } = await spwanTest("req", ["--ci"]);
    expect(code).toEqual(0);
    expect(stdout).toMatchSnapshot();
  });
  test("mock", async () => {
    const { stdout, code } = await spwanTest("req/mock.jsona", ["--ci"]);
    expect(code).toEqual(1);
    expect(stdout).toMatchSnapshot();
  });
  test("trans", async () => {
    const { code } = await spwanTest("req/trans.jsona", ["--ci"]);
    expect(code).toEqual(0);
  });
});
