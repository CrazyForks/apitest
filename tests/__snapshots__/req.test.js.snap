// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`req eval 1`] = `
"main
  @eval complex expr ✔
  fail# @eval syntax error ✘
  fail# @eval must be string value ✘

1. fail# @eval syntax error(main.test2)
   main.test2.req.v1@eval: throw err, Unexpected identifier

2. fail# @eval must be string value(main.test3)
   main.test3.req.v1@eval: should have string value

"
`;

exports[`req file 1`] = `
"main
  @file ✔
  @file ✘

1. @file(main.test2)
   main.test2.req.v1@file: cannot read file

"
`;

exports[`req main 1`] = `
"main
  req variables ✔
  all kinds ✔
  nest value ✔
"
`;

exports[`req mock 1`] = `
"main
  @mock ✘
  fail# @mock ✘
  fail# @mock must be string value ✘

1. @mock(main.test1)
   main.test1.req.v1@mock: bad mock '@username'

2. fail# @mock(main.test2)
   main.test2.req.v1@mock: bad mock 'xyz'

3. fail# @mock must be string value(main.test3)
   main.test3.req.v1@mock: should have string value

"
`;
